Sources/TuistCore/Utils/XcodeBuildLocator.swift

```diff
@@ -0,0 +1,77 @@
+import Foundation
+import TSCBasic
+import TuistGraph
+import TuistSupport
+
+public protocol XcodeBuildLocating {
+
+    /// Locates the build output directory for `xcodebuild` command.
+    ///
+    /// - Parameters:
+    ///   - platform: The platform for the built scheme.
+    ///   - projectPath: The path of the Xcode project or workspace.
+    ///   - configuration: The configuration name, i.e. `Release`, `Debug`, or something custom.
+    func locateBuildDirectory(
```

Comment:
I'd unit test this method:
- Returns the right path when the platform is macOS
- Returns the right path when the platform is not macOS